File: src/main/scala/com/reddit/recommender/api/ApifyConfig.scala
package com.reddit.recommender.api

import scala.io.Source

class ApifyConfig(private val apiToken: String, private val actorId: String) {

  def this() = this({
    val (token, _) = ApifyConfig.loadFromEnv()
    token
  }, {
    val (_, actor) = ApifyConfig.loadFromEnv()
    actor
  })

  def getApiToken: String = apiToken

  def getActorId: String = actorId

  def getApiActorId: String = actorId.replace("/", "~")
}

object ApifyConfig {

  private def loadFromEnv(): (String, String) = {
    var source: Source = null
    try {
      source = Source.fromFile(".env")
      val lines = source.getLines().map(_.trim).toList

      val token = findValue(lines, "APIFY_API_TOKEN")
      val actor = findValue(lines, "APIFY_REDDIT_ACTOR_ID")

      if (token.isEmpty || actor.isEmpty) {
        throw new RuntimeException("Missing APIFY_API_TOKEN or APIFY_REDDIT_ACTOR_ID in .env file")
      }

      (token, actor)
    } catch {
      case e: Exception =>
        throw new RuntimeException("Failed to load .env file: " + e.getMessage, e)
    } finally {
      if (source != null) {
        source.close()
      }
    }
  }

  private def findValue(lines: List[String], key: String): String = {
    val prefix = key + "="
    lines
      .find { line => line.startsWith(prefix) }
      .map { line => line.substring(prefix.length) }
      .getOrElse("")
  }
}

File: src/main/scala/com/reddit/recommender/api/ApifyConfig.java
//package com.reddit.recommender.api;
//
//import java.nio.file.Files;
//import java.nio.file.Paths;
//import java.util.stream.Collectors;
//
//public class ApifyConfig {
//    private String apiToken;
//    private String actorId;
//
//    public ApifyConfig() {
//        loadFromEnv();
//    }
//
//    public ApifyConfig(String apiToken, String actorId) {
//        this.apiToken = apiToken;
//        this.actorId = actorId;
//    }
//
//    private void loadFromEnv() {
//        try {
//            String envContent = Files.lines(Paths.get(".env"))
//                    .collect(Collectors.joining("\n"));
//
//            this.apiToken = extractValue(envContent, "APIFY_API_TOKEN");
//            this.actorId = extractValue(envContent, "APIFY_REDDIT_ACTOR_ID");
//
//            if (apiToken == null || actorId == null) {
//                throw new RuntimeException("Missing APIFY_API_TOKEN or APIFY_REDDIT_ACTOR_ID in .env");
//            }
//
//        } catch (Exception e) {
//            throw new RuntimeException("Failed to load configuration: " + e.getMessage(), e);
//        }
//    }
//
//    private String extractValue(String envContent, String key) {
//        if (envContent == null) return null;
//        for (String line : envContent.split("\n")) {
//            if (line.startsWith(key + "=")) {
//                return line.substring(key.length() + 1).trim();
//            }
//        }
//        return null;
//    }
//
//    // Getters
//    public String getApiToken() { return apiToken; }
//    public String getActorId() { return actorId; }
//    public String getApiActorId() { return actorId.replace("/", "~"); }
//
//
//    // no setters
//
//}